### ISSUE TITLE
Downloaded gps fixes have all a date of 1970/1/1

### ISSUE THREAD

**Issue by nqdrizzt** [subsurface/subsurface#567]
![screenshot_20170912-173526](https://user-images.githubusercontent.com/31509303/30334950-5fa524e2-97e1-11e7-8bc8-4d66b375267d.png)

I am running the GPS service on one device and recorde the fixes to map them later on another device where the DC is connected. The recorded list of fixes is then uploaded manually to the cloud. But after the download of these GPS fixes from the cloud to the other device, the list of GPS fixes have all a timestamp from 1970/1/1. Therefore they can't be mapped to any dives downloaded from the DC into the app.  Please see attached screenshot

Tested with version 4.6.4.759 on subsurface mobile on Android

**Comment by dirkhh** [comment]
That's odd. So just to make sure I understand - you are running Subsurface-mobile on both devices? They are both on Android? They are both connected to the same account, but during the upload to the GPS/Webservice the date part of the time stamp gets lost?

**Comment by nqdrizzt** [comment]
Yes, that's absolutely correct. I also realized that there are error messages in the log returned by the server during uploading the GPS fixes via the API. Please see attached log file generated on the device, which collected and uploaded the GPS fixes. 

[subsurface.txt](https://github.com/Subsurface-divelog/subsurface/files/1302598/subsurface.txt)

**Comment by dirkhh** [comment]
INTERNAL SERVER ERROR - darn, I wish we printed a better error message...
The backend is one of the few pieces of our infrastructure that I am not all that comfortable with (the running joke is that it's in Python and Python and I don't get along). I'll take a look to try and figure out what's going on there because the data that you are sending seems legit at first glance...
I assume that you can reproduce the issue, so if you try again, you get the same result?

**Comment by nqdrizzt** [comment]
Yes I can confirm that it's reproduceable any time. Du u need any further loggings?

**Comment by dirkhh** [comment]
OK, so the errors are completely misleading - they are most likely just "duplicate entries". What I still don't understand is why for some of the GPS fixes the date is wrong.
When I look at the backend database there are indeed entries with 1970 as data (and others with correct looking dates). And for many fixes that I've checked, there were identical entries both for 1970 and 2017 with matching dive time, lat and long.
I literally have no idea what's going on there.
One thing I notice, though. For example the last entry in your screen shot. 1:24pm, -8.36032/116.087 there is no such matching entry - just one from 1970. Yet the log that you attached shows no attempt to send such a fix.
Any idea why that might be?

**Comment by nqdrizzt** [comment]
The log I attached was done at a later point of time as the screenshot. In between I manually tried to explicitly delete all entries with a timestamp of 1970 via the app and also cleared the GPS fix cache on the device. Probably the deletion of that fixes also wiped them from database and at the time of the log file creation they were already gone.  About The duplicate entries (one from 1970 and 2017)I don't have any clue . Maybe  because I mapped the dives manually to a location before uploading them to the backend? Some of the dives I also edited in the windows version of subsurface. But these have to be March 2017 or prior  (not that ones from this week)
